### DISCUSSION TITLE
Benutzerdefinierte Werkzeuge anzeigen

### DISCUSSION THREAD

**Discussion by thomaskofler** [e-netze/webgis-community#95] [General]
Wir hätten den Wunsch, dass man im map builder die benutzerdefinierten Werkzeuge auswählen kann, ohne ein zusätzliches Werkzeug auswählen zu müssen. Momentan werden die benutzerdefinierten Werkzeuge nur angezeigt, wenn man ein standardisiertes Werkzeug z.B. Strecke messen auswählt.

**Comment by gis-eni** [comment]
Benutzerdefinierte Werkzeuge werden in der ``custom.js`` definiert. Diese wird automatisch in die Karte geladen, nicht aber (zwingend) in den MapBuilder. Daher werden sie dort auch nicht angezeigt.

**Reply by thomaskofler** [reply]
Wenn man kein Werkzeug im map builder auswählt, dann erscheinen auch die benutzerdefinierten Werkzeuge nicht (auch wenn man sie in der customjs. in einer Karte definiert):

Ohne Werkzeugauswahl im map builder:
![grafik](https://github.com/gis-eni/webgis-forum/assets/119412172/0523d5e9-c27f-4e81-ae5a-972eb6075c57)

Mit Werkzeugauswahl im map builder:
![grafik](https://github.com/gis-eni/webgis-forum/assets/119412172/c4237d53-4812-4802-8e61-8276c262ed37)

**Comment by gis-eni** [comment]
Geht es hier "nur" um dem MapBuilder, oder möchtest du einfach eine Karte für den Anwender bauen, die nur eine benutzerdefiniertes Werkzeug enthält. Und das wird dann nicht in der Karte (für den Anwender) angezeigt.

Wenn es nur um den MapBuilder geht:
Ist so. MapBuilder gibt nur eine Vorschau wieder. Nicht alle Werkzeugt funktionieren im MapBuilder. Um eine Karte/Werkzeuge zu testen muss immer die Karte selbst aufgerufen werden.

**Reply by thomaskofler** [reply]
Im map builder wäre es mir ja gleich, wenn die Werkzeuge nicht angezeigt werden, aber wenn ich die Karte veröffentliche, dann werden die auch nicht angezeigt:
![grafik](https://github.com/gis-eni/webgis-forum/assets/119412172/0e5ea1cc-d12e-4584-aeb5-6aca89c3cc51)

**Reply by thomaskofler** [reply]
Im Vergleich dazu, wenn ich zB Strecke messen aktiviere, dann erscheinen auch die benutzerdefinierten Werkzeuge:
![grafik](https://github.com/gis-eni/webgis-forum/assets/119412172/1f898c68-050e-4283-9b5a-08021534650d)

**Comment by gis-eni** [comment]
Ok, wenn es um die Werkzeuge geht, dass könnte man das aus "Bug" durchgehen lassen. 
Kann ich mir anschauen, wird aber leider dauern :(

**Comment by gis-eni** [comment]
Problem ist mit Build 5.23.3601 behoben. Benutzerdefinierte Werkzeuge werden auch angezeigt, wenn sich keine Standard Werkzeuge in der Karte oder in einem Container befindet.

Für Container kann jetzt auch eine beliebiger Name angegeben werden, zB `Metadata`:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/d40bc971-92bc-44ae-9eb0-54778ee48b54)
